crates/plugins/mediator-coordination/src/plugin.rs

```diff
@@ -86,42 +82,45 @@ impl Plugin for MediatorCoordinationPlugin {
     }
 
     fn routes(&self) -> Router {
-        let msg = "This should not occur following successful mounting.";
+        // Ensure the plugin is properly mounted
+        let env = self.env.as_ref().expect("Plugin not mounted");
+        let db = self.db.as_ref().expect("Plugin not mounted");
 
-        let storage_dirpath = std::env::var("SERVER_PUBLIC_DOMAIN").unwrap();
-        let mongo_uri = std::env::var("MONGO_URI").unwrap();
-        let mongo_dbn = std::env::var("MONGO_DBN").unwrap();
-        let public_domain = std::env::var("MONGO_DBN").unwrap();
+        let msg = "This should not occur following successful mounting.";
 
         // Load crypto identity
         let mut fs = StdFileSystem;
-        let diddoc = util::read_diddoc(&fs, &storage_dirpath).expect(msg);
-        let keystore = util::read_keystore(&mut fs, &storage_dirpath).expect(msg);
-
-        // Load connection to database
-        let db = load_mongo_connector(&mongo_uri, &mongo_dbn).expect(msg);
+        let diddoc = util::read_diddoc(&fs, &env.storage_dirpath).expect(msg);
+        let keystore = util::read_keystore(&mut fs, &env.storage_dirpath).expect(msg);
```

Comment:
I think you should also do this in the mount method. What do you think?

Reply:
That’s a good suggestion , but moving it to the mount method would require some refactoring of the codebase. Since the current approach isn’t causing any issues, it might be better to leave it as is for now.